pandas/core/apply.py

```diff
@@ -1083,23 +1083,28 @@ def apply(self) -> DataFrame | Series:
     def agg(self):
         result = super().agg()
         if result is None:
-            f = self.f
-
             # string, list-like, and dict-like are entirely handled in super
+            obj = self.obj
+            f = self.f
             assert callable(f)
 
-            # try a regular apply, this evaluates lambdas
-            # row-by-row; however if the lambda is expected a Series
-            # expression, e.g.: lambda x: x-x.quantile(0.25)
-            # this will fail, so we can try a vectorized evaluation
-
-            # we cannot FIRST try the vectorized evaluation, because
-            # then .agg and .apply would have different semantics if the
-            # operation is actually defined on the Series, e.g. str
+            # GH53325: The setup below is just to keep current behavior while emitting a
+            # deprecation message. In the future this will all be replaced with a simple
+            # `result = f(self.obj)`.
+            if isinstance(f, np.ufunc):
+                with np.errstate(all="ignore"):
+                    return f(obj)
             try:
-                result = self.obj.apply(f)
+                result = obj.apply(f)
             except (ValueError, AttributeError, TypeError):
                 result = f(self.obj)
+            else:
+                msg = (
+                    f"using {f} in {type(obj).__name__}.agg cannot aggregate and "
+                    f"has been deprecated. Use {type(obj).__name__}.transform to "
+                    f"keep behavior unchanged."
+                )
+                warnings.warn(msg, FutureWarning, stacklevel=find_stack_level())
```

Comment:
This will warn on 

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': [3, 4, 5]})
df.apply([lambda x: x])
```

Reply:
Wow, that is surprising...and annoying. I'll se what I can do about that.

Reply:
If you can find some way, it'd be great. I worked on this for a while and gave up. This is where I think we need an option to opt into a whole new implementation (the HOMs API PDEP that I'm slowly working on).